### PR TITLE
feat: post hook in fal run

### PR DESCRIPTION
**Pull request by mederka**
This is only for `fal run`. For `fal flow run` see [here](#386)

With this yaml:

```
version: 2

models:
  - name: model_a
    meta:
      fal:
        post-hook:
          - hook.py
          - hook2.py
```

```
dbt run
fal run
```

Will run hook.py and hook2.py. Both `write_to_source` and `write_to_model` are not allowed in these.

Closes FEA-177
Closes FEA-167

Question for @chamini2, @burkaygur and @turbo1912:

Should we allow script selection for post-hooks?

### CODE REVIEW

File: src/fal/cli/fal_runner.py [hunk 1]
```diff
@@ -75,6 +79,16 @@ def _scripts_flag(args: argparse.Namespace) -> bool:
     return bool(args.scripts)
 
 
+def _get_all_post_hooks(models: List[DbtModel], faldbt: FalDbt):
```
Comment (turbo1912) [inline src/fal/cli/fal_runner.py:82]: return type here

File: src/fal/cli/fal_runner.py [hunk 2]
```diff
@@ -75,6 +79,16 @@ def _scripts_flag(args: argparse.Namespace) -> bool:
...
+    return list(chain(*(_get_model_post_hooks(model, faldbt) for model in models)))
+
+
+def _get_model_post_hooks(model: DbtModel, faldbt: FalDbt) -> List[FalScript]:
```
Comment (turbo1912) [inline src/fal/cli/fal_runner.py:86]: suggestion for method name `_get_post_hooks_for_model`

File: src/fal/fal_script.py [hunk 3]
```diff
@@ -64,12 +64,20 @@ class FalScript:
...
-
-    def __init__(self, faldbt: FalDbt, model: Optional[DbtModel], path: str):
+    _is_post_hook: bool
+
+    def __init__(
+        self,
+        faldbt: FalDbt,
+        model: Optional[DbtModel],
+        path: str,
+        is_post_hook: bool = False,
```
Comment (turbo1912) [inline src/fal/fal_script.py:74]: Add a todo here that we wont need this property when after scripts are deprecated

File: src/fal/fal_script.py [hunk 4]
```diff
@@ -210,3 +227,15 @@ def strip_magic(source: List[str]) -> List[str]:
     logger.debug(f"Joined .ipynb cells to:\n{joined_script}")
 
     return joined_script
+
+
+def _not_allowed_function_maker(function_name: str):
```
Comment (turbo1912) [inline src/fal/fal_script.py:232]: return type here

**Review by turbo1912 — APPROVED** [review]
small nitpick comments nothing blocking

File: src/fal/fal_script.py [hunk 5]
```diff
@@ -210,3 +229,15 @@ def strip_magic(source: List[str]) -> List[str]:
     logger.debug(f"Joined .ipynb cells to:\n{joined_script}")
 
     return joined_script
+
+
+def _not_allowed_function_maker(function_name: str) -> Callable[[Any], None]:
```
Comment (chamini2) [inline src/fal/fal_script.py:234]: good one

**Pull request merged**